pep-0722.rst

```diff
@@ -0,0 +1,271 @@
+few more experienced developers) will often *start* a project by just opening up
+an editor and writing a simple script. In some cases, the project might outgrow
+the "simple script" phase and get restructured into a more conventional project
+directory, but not all do, and even those that do still need to be usable in the
+"simple script" phase.
+
+These days, the idea that "simple scripts can just use the stdlib" is becoming
+less and less practical, as the quantity, quality and usability of libraries on
+PyPI steadily increases. For example, using the stdandard ``urllib`` library
+rather than something like ``requests`` or ``httpx`` makes correct code
+significantly harder to write.
+
+Having to consider "uses 3rd party libraries" as the break point for moving to a
+"full scale project" is impractical, so this PEP is designed to allow a project
+to use external libraries while still remaining as a simple, standalone script.
+
+Of course, *declaring* your dependencies isn't sufficient by itself. You need to
+install them (probably in some sort of virtual environment) so that they are
+available when the script runs. This PEP does not cover environment management,
+as tools like `pip-run <https://pypi.org/project/pip-run/>`__ and `pipx
+<https://pypi.org/project/pipx/>`__ already offer that ability. But by
+standardising the means of declaring dependencies, this PEP allows scripts to
+remain tool-independent.
+
+
+Specification
+=============
+
+Any Python script may contain a *dependency block*, which is a specially
+structured comment block. The format of the dependency block is as follows:
+
+* A single comment line containing the (case sensitive) text "Requirements:"
+* A series of comment lines containing :pep:`508` requirement specifiers.
+* An empty comment or blank line.
+
+To be recognised as a "comment line", the line must start with a ``#`` symbol.
+Leading and trailing whitespace on the comment line is ignored.
+
+The dependency block may occur anywhere in the file. There MUST only be a single
+dependency block in the file - tools consuming dependency data MAY simply
+process the first dependency block found. This avoids the need for tools to
+process more data than is necessary. Stricter tools MAY, however, fail with an
+error if multiple dependency blocks are present.
+
+Example
+-------
+
+The following is an example of a script with an embedded dependency block::
+
+    #!/usr/bin/env python
+
+    # In order to run, this script needs the following 3rd party libraries
+    #
+    # Requirements:
+    #    requests
+    #    rich
+
+    import requests
+    from rich.pretty import pprint
+
+    resp = requests.get("https://peps.python.org/api/peps.json")
+    data = resp.json()
+    pprint([(k, v["title"]) for k, v in data.items()][:10])
+
+
+Backwards Compatibility
+=======================
+
+As the dependency data is recorded in the form of a structured comment, this is
+compatible with any existing code.
```

Comment:
Perhaps stretching the bounds of plausibility, but if someone has a file with exactly a structure as defined in this PEP, tools could unexpectedly start installing packages when the script runs. A potential compatability concern, though I don't think particularly major.